docs/content/develop/resource.md

```diff
@@ -216,6 +216,11 @@ For more information about types of resources and the generation process overall
 
 ## Add fields
 
+In general, Terraform resources should implement all configurable fields, as well as all
```

Comment:
We should probably just lean towards all fields nowadays. That's gonna be the easiest to automate / do coverage detection on, and we've seen old fields we excluded get requested in older products.

Reply:
made that the default and indicated when it's okay to deviate.